integration-tests/smoke/ccip/ccip_rmn_test.go

```diff
@@ -448,9 +404,241 @@ type rmnTestCase struct {
+			activeDigest[:], config.Config.RmnHomeContractConfigDigest[:])
+
+		t.Logf("RMNRemote config digest after setting: %x", config.Config.RmnHomeContractConfigDigest[:])
+	}
+}
+
+func (tc rmnTestCase) killMarkedRmnNodes(t *testing.T, rmnCluster devenv.RMNCluster) {
+	for _, n := range tc.rmnNodes {
+		if n.forceExit {
+			t.Logf("Pausing RMN node %d", n.id)
+			rmnN := rmnCluster.Nodes["rmn_"+strconv.Itoa(n.id)]
+			require.NoError(t, osutil.ExecCmd(zerolog.Nop(), "docker kill "+rmnN.Proxy.ContainerName))
+			t.Logf("Paused RMN node %d", n.id)
+		}
+	}
+}
+
+func (tc rmnTestCase) disableOraclesIfThisIsACursingTestCase(ctx context.Context, t *testing.T, envWithRMN changeset.DeployedEnv) []string {
+	disabledNodes := make([]string, 0)
+
+	if len(tc.cursedSourceChainIdxs) > 0 || tc.globalCurse {
+		listNodesResp, err := envWithRMN.Env.Offchain.ListNodes(ctx, &node.ListNodesRequest{})
+		require.NoError(t, err)
+
+		for _, n := range listNodesResp.Nodes {
+			if strings.HasPrefix(n.Name, "bootstrap") {
+				continue
+			}
+			_, err := envWithRMN.Env.Offchain.DisableNode(ctx, &node.DisableNodeRequest{Id: n.Id})
+			require.NoError(t, err)
+			disabledNodes = append(disabledNodes, n.Id)
+			t.Logf("node %s disabled", n.Id)
+		}
+	}
+
+	return disabledNodes
+}
+
+func (tc rmnTestCase) sendMessages(t *testing.T, onChainState changeset.CCIPOnChainState, envWithRMN changeset.DeployedEnv) (map[uint64]*uint64, map[changeset.SourceDestPair]uint64, map[changeset.SourceDestPair][]uint64) {
+	startBlocks := make(map[uint64]*uint64)
+	seqNumCommit := make(map[changeset.SourceDestPair]uint64)
+	seqNumExec := make(map[changeset.SourceDestPair][]uint64)
+
+	for _, msg := range tc.messagesToSend {
+		fromChain := tc.pf.chainSelectors[msg.fromChainIdx]
+		toChain := tc.pf.chainSelectors[msg.toChainIdx]
+
+		for i := 0; i < msg.count; i++ {
+			msgSentEvent := changeset.TestSendRequest(t, envWithRMN.Env, onChainState, fromChain, toChain, false, router.ClientEVM2AnyMessage{
+				Receiver:     common.LeftPadBytes(onChainState.Chains[toChain].Receiver.Address().Bytes(), 32),
+				Data:         []byte("hello world"),
+				TokenAmounts: nil,
+				FeeToken:     common.HexToAddress("0x0"),
+				ExtraArgs:    nil,
+			})
+			seqNumCommit[changeset.SourceDestPair{
+				SourceChainSelector: fromChain,
+				DestChainSelector:   toChain,
+			}] = msgSentEvent.SequenceNumber
+			seqNumExec[changeset.SourceDestPair{
+				SourceChainSelector: fromChain,
+				DestChainSelector:   toChain,
+			}] = []uint64{msgSentEvent.SequenceNumber}
+			t.Logf("Sent message from chain %d to chain %d with seqNum %d", fromChain, toChain, msgSentEvent.SequenceNumber)
+		}
+
+		zero := uint64(0)
+		startBlocks[toChain] = &zero
+	}
+
+	return startBlocks, seqNumCommit, seqNumExec
+}
+
+func (tc rmnTestCase) callContractsToCurseChains(ctx context.Context, t *testing.T, onChainState changeset.CCIPOnChainState, envWithRMN changeset.DeployedEnv) {
```

Comment:
Can you add a todo here to use a proposal instead, ticket: https://smartcontract-it.atlassian.net/browse/CCIP-4400